common:
  # Topics
  pointCloudTopic:  "/os_cloud_node/points" #"/os_cloud_node/points" #"velodyne_points"    "points_raw"     rslidar_points      # Point cloud data
  imuTopic: "/os_cloud_node/imu" #"/os_cloud_node/imu" #"imu_data" #"imu_raw"          # IMU data


feature_extract:
  lidarFrame: "base_link"

  # Sensor Settings
  sensor: ouster #velodyne                            # lidar sensor type, either 'velodyne' or 'ouster' 'robosense' 'livox'
  N_SCAN: 32                                  # number of lidar channel (i.e.,Velodyne/Ouster/Robosense: 16, 32, 64, 128,Livox Horizon:6)
  Horizon_SCAN: 1024                          # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048,Livox Horizon: 4000)
  downsampleRate: 1                           # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1
  lidarMinRange: 1.0                          # default: 1.0, minimum lidar range to be used
  lidarMaxRange: 1000.0                       # default: 1000.0, maximum lidar range to be used

  # LOAM feature threshold
  edgeThreshold: 0.3
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100

  # voxel filter paprams
  odometrySurfLeafSize: 0.5                     # default: 0.4 - outdoor, 0.2 - indoor  only use surf ds here


mapping:
  IMU_Mode: 2    # 0-not use imu, 1-use imu remove rotation distort, 2-tightly coupled imu
  filter_parameter_corner: 0.2  # Voxel Filter Size Use to Downsize Map Cloud
  filter_parameter_surf: 0.5
  extrinsic_T: [ 0, 0, 0.0] # lidar to imu
  extrinsic_R: [ 1, 0, 0, 
                 0, 1, 0, 
                 0, 0, 1]


location:
  filedir: "/home/cc/catkin_velodyne/src/LIO_Localization/datamap"
  pointCloudTopic: "/laser_cloud_filtered"
  imuTopic: "/os_cloud_node/imu"
  IMU_Mode: 1
  corner_leaf_: 0.2
  surf_leaf_: 1.0